pkg/provider/kubernetes/crd/kubernetes.go

```diff
@@ -712,6 +714,24 @@ func (p *Provider) createErrorPageMiddleware(client Client, namespace string, er
 	return errorPageMiddleware, balancerServerHTTP, nil
 }
 
+func (p *Provider) FillExtensionBuilderRegistry(registry gateway.ExtensionBuilderRegistry) {
+	registry.RegisterFilterFuncs(traefikv1alpha1.GroupName, "Middleware", func(name, namespace string) (string, *dynamic.Middleware, error) {
+		if len(p.Namespaces) > 0 && slices.Contains(p.Namespaces, namespace) {
```

Comment:
```suggestion
		if len(p.Namespaces) > 0 && !slices.Contains(p.Namespaces, namespace) {
```
I guess it can be a good idea to add a test on this